storage/static.go

```diff
@@ -36,25 +32,50 @@ func (s staticClientsStorage) GetClient(id string) (Client, error) {
 	if client, ok := s.clientsByID[id]; ok {
 		return client, nil
 	}
-	return Client{}, ErrNotFound
+	return s.Storage.GetClient(id)
+}
+
+func (s staticClientsStorage) isStatic(id string) bool {
+	_, ok := s.clientsByID[id]
+	return ok
 }
 
 func (s staticClientsStorage) ListClients() ([]Client, error) {
-	clients := make([]Client, len(s.clients))
-	copy(clients, s.clients)
-	return clients, nil
+	clients, err := s.Storage.ListClients()
+	if err != nil {
+		return nil, err
+	}
+	n := 0
+	for _, client := range clients {
+		// If a client in the backing storage has the same ID as a static client
+		// prefer the static client.
+		if !s.isStatic(client.ID) {
+			clients[n] = client
+			n++
+		}
+	}
+	return append(clients[:n], s.clients...), nil
 }
 
 func (s staticClientsStorage) CreateClient(c Client) error {
-	return errors.New("static clients: read-only cannot create client")
+	if s.isStatic(c.ID) {
+		return errors.New("static clients: read-only cannot create client")
```

Comment:
I don't like adding yet another sentinel error :( Maybe we could introduce a `storage.Error` interface instead? Something like

```go
type ErrKind int

const (
    ErrAlreadyExists ErrKind = iota
    ErrNotFound
    ErrNotModifiable
    ErrUnknown
)

type Error interface {
    Kind() ErrKind
}
```

Then use that to do all of this custom detection?